src/Microsoft.Restier.Publishers.OData/RestierController.cs

```diff
@@ -525,6 +578,127 @@ private ODataPath GetPath()
             return originalValues;
         }
 
+        private async Task<IQueryable> ExecuteUnboundFunction(UnboundFunctionPathSegment segment,
```

Comment:
The major part of `ExecuteUnboundFunction` and `ExecuteBoundFunction` is the same. The difference is how we handle the first parameter (the binding parameter). Consider extract the common logic and make it clear, e.g., `GetMethod`, `ReadBindingParameter` (optional for unbound), `ReadParameters`, `ExecuteFunction` (suggest renaming it to more precise `InvokeMethod`).